chutney/ui/src/app/shared/components/execute-modal/scenario-execute-modal.component.ts

```diff
@@ -39,27 +50,82 @@ export class ScenarioExecuteModalComponent implements OnInit {
     constructor(
         private datasetService: DataSetService,
         private environmentService: EnvironmentService,
-        private eventManagerService: EventManagerService,
-        private translateService: TranslateService) {
+        private formBuilder: FormBuilder,
+        private translateService: TranslateService,
+        private changeDetectorRef: ChangeDetectorRef){
     }
 
 
     ngOnInit(): void {
-        this.datasetService.findAll().subscribe((res: Array<Dataset>) => {
-            this.datasets = res;
-            this.filteredDatasets = res;
+        this.translateService.get('dataset.actions.createDataset').pipe(
```

Comment:
You can use sync method `translateService.instant`. It avoids having an observable inside an other.
otherwise, you can use `observable.switchmap` if you prefer to use `translationservice.get`